### DISCUSSION TITLE
Use coo_array/ndarray with MKL

### DISCUSSION THREAD

**Discussion by bentrainer** [wjakob/nanobind#763] [Q&A]
Hi, this may be a stupid question because I just started to learn C++ 😭.

I want to achieve a simple task using nanobind to call `mkl_sparse_z_trsm` in Python so I could solve `Ax=b` where `A` is a sparse upper triangular matrix (`scipy.sparse.coo_array`) while `b` is a dense matrix (`np.ndarray`).

I have not dig into `coo_array` but stuck at the `b` part with `foo(nb::ndarray<>& b)`. Because it's an complex array, I don't know how to convert it into `MKL_Complex16` type even though I think the `numpy` also use mkl to store the underlying data. I tried to pass the `b.data()` into the mkl function but got:
```
argument of type "nanobind::ndarray<>::Scalar *" (aka "void *") is incompatible with parameter of type "MKL_Complex16 *" (aka "_MKL_Complex16 *")
```

What is the correct way to convert it to a `MKL_Complex16` pointer?

BTW, the reason I want to try this rather than simply go `spsolve_triangular` or `spsolve` from `scipy.sparse.linalg` is because they are way slow compared to MATLAB. I want to figure out where the performance goes...

**Comment by wjakob** [comment]
Can you use a `static_cast`?
